Add aot_inductor as a test backend for benchmarking

desertfire opened the pull request:
Summary:
Original PR at https://github.com/pytorch/pytorch/pull/104977. Landing from fbcode instead.

Add an aot_inductor backend (Export+AOTInductor) in the benchmarking harness. Note it is not a dynamo backend.

Moved files from torch/_inductor/aot_inductor_include to torch/csrc/inductor as a more standard way for exposing headers
Created a caching function in benchmarks/dynamo/common.py for compiling, loading and caching the .so file, as a proxy for a pure C++ deployment, but easier for benchmarking.

Differential Revision: D47452591



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @anijain2305

jansel commented on benchmarks/dynamo/common.py:
```diff
@@ -2716,7 +2803,7 @@ def get_example_inputs(self):
     group.add_argument(
         "--backend",
-        choices=torch._dynamo.list_backends(exclude_tags=None),
+        choices=torch._dynamo.list_backends(exclude_tags=None) + ["aot_inductor"],
```
That does seem cleaner.  Perhaps we could also just use the standard backened registration mechanism.

desertfire replied:
I intentionally avoid doing that because I think aot_inductor is not a dynamo backend. Maybe we should add a backend registration for export.

desertfire replied:
I will just make it a separate option for now, similar to [--export](https://github.com/pytorch/pytorch/blob/e68cf02420b852c3bf278fe9d1f47d6b02bae0f0/benchmarks/dynamo/common.py#L2698). An export backend registration can come next.

desertfire committed:
Add aot_inductor as a test backend for benchmarking (#105221)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/105221

Original PR at https://github.com/pytorch/pytorch/pull/104977. Landing from fbcode instead.

Add an aot_inductor backend (Export+AOTInductor) in the benchmarking harness. Note it is not a dynamo backend.

Moved files from torch/_inductor/aot_inductor_include to torch/csrc/inductor as a more standard way for exposing headers
Created a caching function in benchmarks/dynamo/common.py for compiling, loading and caching the .so file, as a proxy for a pure C++ deployment, but easier for benchmarking.

Reviewed By: chenyang78

Differential Revision: D47452591

fbshipit-source-id: b8eb914a9d1c77037c4ea3da51e36c7f2ac70726
